include/swift/AST/DiagnosticsSema.def

```diff
@@ -2069,6 +2069,14 @@ NOTE(codable_codingkeys_type_is_not_an_enum_here,none,
      "cannot automatically synthesize %0 because 'CodingKeys' is not an enum", (Type))
 NOTE(codable_codingkeys_type_does_not_conform_here,none,
      "cannot automatically synthesize %0 because 'CodingKeys' does not conform to CodingKey", (Type))
+NOTE(decodable_no_super_init_here,none,
+     "cannot automatically synthesize %0 because superclass does not have a callable 'init()' or 'init(from:)'", (DeclName))
+NOTE(decodable_super_init_not_designated_here,none,
+     "cannot automatically synthesize %0 because required superclass initializer %1 is not designated", (DeclName, DeclName))
```

Comment:
To go along with the comment on the other diagnostics — what I'm trying to get across here (perhaps poorly) is that the implementation of `init(from:)` we'd generate needs to be able to call this member. Trying to explain this in a diagnostic gets a bit wordy, so I settled on "required" because it appeared in other diagnostics.

Another option I tried (but opted not to use) was `cannot synthesize %0 because implementation would need to call %1, which is not designated` (or similar). Does that sound more reasonable, or should I find a different way to rephrase?